src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -297,24 +287,27 @@
  * You can create the {@code ChoiceFormat} programmatically, as in the
  * above example, or by using a pattern. See {@link ChoiceFormat}
  * for more information.
- * <blockquote>
  * {@snippet lang=java :
- * form.applyPattern(
- *    "There {0,choice,0#are no files|1#is one file|1<are {0,number,integer} files}.");
+ * msgFmt.applyPattern(
+ *    "There {0,choice,0#are no files|1#is one file|1<are {1,number,integer} files}.");
  * }
- * </blockquote>
  *
  * <p>
- * <strong>Note:</strong> As we see above, the string produced
- * by a {@code ChoiceFormat} in {@code MessageFormat} is treated as special;
- * occurrences of '{' are used to indicate subformats, and cause recursion.
+ * <strong id="pattern_caveats">Notes:</strong> As seen in the previous snippet,
+ * the string produced by a {@code ChoiceFormat} in {@code MessageFormat} is
+ * treated as special; occurrences of '{' are used to indicate subformats, and
+ * cause recursion. A limitation to this behavior is that, if a {@code MessageFormat}
```

Comment:
@naotoj Please let me know if the added limitations wording reads clearly to you. I would rather not have a whole dedicated code example to explain a _limitation_, but if needed for better understanding, I can add one.

Reply:
Do you mean `{@code MessageFormat}{@code FormatElement}`?
Also, I would rather remove "A limitation to this behavior is that," as readers would know the following explanation explains a limitation.

Reply:
Right, no need for `{@code MessageFormat}` as it's already implied. Also removed "_A limitation to this behavior is that_" as suggested.